### ISSUE TITLE
Ability to record voice messages on a contact record

### ISSUE THREAD

**Issue by corsacca** [DiscipleTools/disciple-tools-theme#2001]
![image](https://user-images.githubusercontent.com/24901539/227463512-60577a52-4969-49d2-9248-3ebc44601ed2.png)

Bonus: auto transcripts.

**Comment by corsacca** [comment]
Notes:
One of the big issues is storage of pictures and audio.
The uploads folder does not feel very secure and most hosting don't give a lot of space.
This highlights a need for media to be able to be stored in s3 or backblase or similar. with the links to the audio stored in D.T

**Comment by cairocoder01** [comment]
To support various cloud storage providers, it would be smart to setup plenty of hooks in order to allow custom plugins to support other providers. I did this with my data reporting plugin (https://github.com/cairocoder01/disciple-tools-data-reporting), so that might be worth looking at. It provides basic HTTP functionality and then there are 2 plugins that create providers for [Google](https://github.com/cairocoder01/disciple-tools-data-reporting-provider-gcp) and [Azure](https://github.com/cairocoder01/disciple-tools-data-reporting-provider-azure).

**Comment by corsacca** [comment]
Yes! hook would be great so we can build this in collaboration with this feature: https://github.com/DiscipleTools/disciple-tools-theme/issues/2395
